__tests__/accessiblity-audit.test.js

```diff
@@ -20,58 +16,47 @@ async function audit (page) {
     .exclude('.govuk-skip-link')
     // axe reports that the back to top button is not inside a landmark, which is intentional.
     .exclude('.app-back-to-top')
+    // axe reports that the frame "does not have a main landmark" and example <h1> headings
+    // violate "Heading levels should only increase by one", which is intentional.
+    // https://github.com/alphagov/govuk-design-system/pull/2442#issuecomment-1326600528
+    .exclude('.app-example__frame')
```

Comment:
We've had to exclude `iframe` "app example" elements but added a comment

```js
// axe reports that the frame "does not have a main landmark" and example <h1> headings
// violate "Heading levels should only increase by one", which is intentional.
// https://github.com/alphagov/govuk-design-system/pull/2442#issuecomment-1326600528
```

See discussion in https://github.com/alphagov/govuk-design-system/pull/2442#issuecomment-1326600528

Reply:
Should we add an issue for making our accessibility tests better, especially with regards to this?

First off, we're only checking one component page, and now we're going to be ignoring much of the content of that page. As I've mentioned before, it feels like these tests are a bit toothless as a result.

Alternatively, are we able to more specifically target the elements that are causing failures _within_ the iFrames? I recall when I had a brief dig that was a hassle.

We might also need to update our Accessibility Statement if we feel the confusion between heading levels might cause WCAG issues?

Reply:
Hey @domoscargin I'm having a bit of trouble parsing the context from this thread but I'll start by answering your question: I think I've heard from Ollie that in the past, we would ignore the axe warning about having multiple <h1> elements in iframe examples. Due to our nature as a design system which offers coded examples, we don't really have a better option than to use whatever headings are needed in our iframe examples. And due to our context, it is likely to be understood by users.

On another note, dies this thread mean that we've gone from testing the first coded <iframe> example in a page, to not testing any of our coded examples at all? If so, that's not a great direction to be moving. If that's not the case, then I've misinterpreted and carry on 👍

Reply:
I think the testing you're referring to is that on `govuk-frontend`, we [run accessibility tests against the "default" example of each component](https://github.com/alphagov/govuk-frontend/blob/c5fc0e39a70fada4ec527f912ea6fb86aad71150/src/govuk/components/radios/template.test.js#L13-L18), which is pretty much always the first `<iframe>` on a Design System component page.

_This_ testing is currently run only against certain pages on the Design System. The only component page it hits is Radios.

So theoretically it should've been testing the entire Radios page, including the iframes, but there seems to have been a bug in `axe-puppeteer` where iframes were ignored, so we weren't actually testing them at all. 

Now that we've updated to `@axe-core/puppeteer`, we're actually testing the iframes, so we've suddenly surfaced the errors within 'em.

Reply:
Thanks @domoscargin @davidc-gds

Yeah this pull request is more of an "upgrade our old tooling" change to unlock good stuff in future

With the new updates we're able to see that axe reports:

1. `<iframe>` "does not have a main landmark"
2. `<iframe>` "Heading levels should only increase by one"

Since they previously weren't supported, I think we're happy to turn the two new violations off. Thankfully everything else still passes but we may want to expand our Design System website checks further than this list soon:

* [Home page](https://design-system.service.gov.uk)
* [Component page (Radios)](https://design-system.service.gov.uk/components/radios/)
* [Patterns page (Gender or sex)](https://design-system.service.gov.uk/patterns/gender-or-sex/)
* [Get in touch](https://design-system.service.gov.uk/get-in-touch/)
* [Sitemap](https://design-system.service.gov.uk/sitemap/)

I'm presuming the checks are fairly minimal as the bulk of the testing happens on GOV.UK Frontend

Reply:
Thanks both @domoscargin @colinrotherham, that clarifies it a lot. In that case, if it's only 1 page with iframes we'd be checking (iframes on the `Radios` page), and the content in those iframes is all covered by the govuk-frontend testing, then I might recommend we:
- don't test within iframes
- continue checking for h1 and landmark issues, but now it won't look in the iframes

Especially if we're deciding between turning off testing for a handful of iframes, versus turning off h1 and landmark checks for the entire rest of the website.

This might have been the conclusion you already came to, but just wanted to document it 👍